### PR TITLE
Parser Cloud refactor

### PR DESCRIPTION
**Pull request by illusional**
Refactor cloud interactions to CloudHelper + use cloudpathlib.
Preparing for ONT changes

### CODE REVIEW

**Review by violetbrina — APPROVED** [review]
All good. Couple questions and very minor comments. But looks good to merge!

File: sample_metadata/parser/cloudhelper.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        bucket = self.get_gcs_bucket(bucket_name)
+
+        # the next few lines are equiv to `bucket.get_blob(path)`
+        # but without requiring storage.objects.get permission
+        blobs = list(self.gcs_client.list_blobs(bucket, prefix=path))
+        # first where r.name == path (or None)
+        return next((r for r in blobs if r.name == path), None)
+
+    def _list_gcs_directory(self, gcs_path) -> list[str]:
+
```
[the remark this answers is not in the document]
Comment (illusional) [reply sample_metadata/parser/cloudhelper.py:158]: You don't need one for internal methods (starting with an underscore) - and I've sort of intentionally left this blank as I think the method name is clear enough.

[the remark this answers is not in the document]
Comment (vladsaveliev) [reply sample_metadata/parser/cloudhelper.py:158]: Sorry to chime in! Wouldn't `CloudPath(gcs_path).iterdir()` work here?

[the remark this answers is not in the document]
Comment (vladsaveliev) [reply sample_metadata/parser/cloudhelper.py:158]: @illusional Sorry to chime in! Wouldn't `CloudPath(gcs_path).iterdir()` work here?

[the remark this answers is not in the document]
Comment (illusional) [reply sample_metadata/parser/cloudhelper.py:158]: Please always feel free to chime in! Great suggestion, I'll add this in a follow-up PR :)

[the remark this answers is not in the document]
Comment (illusional) [reply sample_metadata/parser/cloudhelper.py:158]: I did some rough testing, and it's super slow - like on the order of minutes slow which is strange?

[the remark this answers is not in the document]
Comment (illusional) [reply sample_metadata/parser/cloudhelper.py:158]: I don't know if I trust their implementation, I feel like it's iterating through everything in the bucket and will only return the correct blobs, but taking a very long time:

https://github.com/drivendataorg/cloudpathlib/blob/20b79c11974fce5b8e7b0fe292ba8d11b00dad49/cloudpathlib/gs/gsclient.py#L140-L172

[the remark this answers is not in the document]
Comment (vladsaveliev) [reply sample_metadata/parser/cloudhelper.py:158]: Huh, yeah, that's a good point. Make sense to stick to the existing approach then :)

**Pull request merged**